internal/indexer/documentation.go

```diff
@@ -0,0 +1,748 @@
+			// separately.
+			for i := range t.Names {
+				switch node.Tok {
+				case token.CONST:
+					constDocs := d.indexConstVar(p, t, i, "const", isTestFile)
+					constDocs.docsMarkdown = blockDocsMarkdown + constDocs.docsMarkdown
+					result.consts = append(result.consts, constDocs)
+				case token.VAR:
+					varDocs := d.indexConstVar(p, t, i, "var", isTestFile)
+					varDocs.docsMarkdown = blockDocsMarkdown + varDocs.docsMarkdown
+					result.vars = append(result.vars, varDocs)
+				}
+			}
+		case *ast.TypeSpec:
+			typeDocs := d.indexTypeSpec(p, t, isTestFile)
+			typeDocs.docsMarkdown = blockDocsMarkdown + typeDocs.docsMarkdown
+			result.types = append(result.types, typeDocs)
+		}
+	}
+	return result
+}
+
+type constVarDocs struct {
+	// The emitted "documentationResult" vertex ID.
+	ID uint64
+
+	// The best one-line label for this type we could come up with, e.g. `var x` omitting
+	// the assignment.
+	label string
+
+	// The name of the const/var.
+	name string
+
+	// The full type signature, with docstrings on e.g. struct fields.
+	signature string
+
+	// Documentation strings in Markdown.
+	docsMarkdown string
+
+	// Is the type itself exported, deprecated?
+	exported, deprecated bool
+}
+
+func (t constVarDocs) result() *documentationResult {
+	var tags []protocol.DocumentationTag
+	if t.exported {
+		tags = append(tags, protocol.DocumentationExported)
+	}
+	if t.deprecated {
+		tags = append(tags, protocol.DocumentationDeprecated)
+	}
+
+	// Include the full type signature
+	var detail bytes.Buffer
+	fmt.Fprintf(&detail, "```Go\n")
+	fmt.Fprintf(&detail, "%s\n", t.signature)
+	fmt.Fprintf(&detail, "```\n\n")
+	fmt.Fprintf(&detail, "%s", t.docsMarkdown)
+
+	return &documentationResult{
+		Documentation: protocol.Documentation{
+			Slug:    slugify(t.name),
+			NewPage: false,
+			Tags:    tags,
+		},
+		Label:  protocol.NewMarkupContent(t.label, protocol.PlainText),
+		Detail: protocol.NewMarkupContent(detail.String(), protocol.Markdown),
+	}
+}
+
+func (d *docsIndexer) indexConstVar(p *packages.Package, in *ast.ValueSpec, nameIndex int, typ string, isTestFile bool) constVarDocs {
+	var result constVarDocs
+	name := in.Names[nameIndex]
+	result.label = fmt.Sprintf("%s %s", typ, name.String())
+	result.name = name.String()
+	result.exported = ast.IsExported(name.String()) && !isTestFile
+	result.deprecated = isDeprecated(in.Doc.Text())
+
+	// Produce the full type signature with docs on e.g. methods and struct fields, but not on the
+	// type itself (we'll produce those as Markdown below.)
+	cpy := *in
+	cpy.Doc = nil
+	result.signature = typ + " " + formatNode(p.Fset, &cpy)
+
+	// TODO(slimsag): future: this is a HACK because some variables/constants are ultra long table
+	// initializers, including those is not helpful to the user so we fallback in this case to
+	// something much briefer.
+	if (strings.HasSuffix(result.signature, "}\n") || strings.HasSuffix(result.signature, "}")) && strings.Contains(result.signature, "{") {
```

Comment:
Could we just check the length instead?

Reply:
I'm going to revisit this later, I don't think spending more time making this specific check smarter (or less smart) makes sense at this point. Rather just fix it for real. :)